homeassistant/components/somfy_mylink/__init__.py

```diff
@@ -136,6 +146,34 @@ def _async_import_options_from_data_if_missing(hass: HomeAssistant, entry: Confi
         hass.config_entries.async_update_entry(entry, data=data, options=options)
 
 
+@callback
+def _async_migrate_entity_config(
+    hass: HomeAssistant, entry: ConfigEntry, mylink_status: dict
+):
+    if CONF_ENTITY_CONFIG not in entry.options:
+        return
+
+    options = dict(entry.options)
+
+    reversed_target_ids = options[CONF_REVERSED_TARGET_IDS] = {}
+    legacy_entry_config = options[CONF_ENTITY_CONFIG]
+
+    for cover in mylink_status["result"]:
+        legacy_entity_id = ENTITY_ID_FORMAT.format(slugify(cover["name"]))
+        target_id = cover["targetID"]
+
+        entity_config = legacy_entry_config.get(legacy_entity_id, {})
+        default_reverse = options.get(CONF_DEFAULT_REVERSE)
```

Comment:
This option doesn't change in the iteration so we can move it up out of the for loop.

Reply:
done